checksum/checksum.go

```diff
@@ -0,0 +1,180 @@
+package checksum
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/schollz/progressbar/v3"
+)
+
+func checksumFor(path string) ([]byte, error) {
+	h := md5.New()
+
+	var r *os.File
+	r, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening file in %s: %w", path, err)
+	}
+	defer r.Close()
+
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("error reading %s checksum content: %w", path, err)
+	}
+
+	return h.Sum(nil), nil
+}
+
+func check(src, target, f string) (bool, error) {
+	name := filepath.Base(f)
+	sc, err := checksumFor(filepath.Join(src, name))
+	if err != nil {
+		return false, fmt.Errorf("error getting src %s checksum: %w", name, err)
+	}
```

Comment:
A mensagem de erro fica mais fácil de entender se utilizarmos o caminho completo (idem para linha 44)

```suggestion
	src = filepath.Join(src, name)
	sc, err := checksumFor(src)
	if err != nil {
		return false, fmt.Errorf("error getting %s checksum: %w", src, err)
	}
```

Aliás, essa função não pode receber apenas 2 argumentos: `src` caminho completo do arquivo, e `target` diretório.